### ISSUE TITLE
Allow other HTTP clients to be used

### ISSUE THREAD

**Issue by mudge** [altmetric/embiggen#9]
Embiggen currently uses Ruby's standard [`Net::HTTP`](http://ruby-doc.org/stdlib-2.2.2/libdoc/net/http/rdoc/Net/HTTP.html) library to perform HTTP requests but this is not open for extension (e.g. to configure proxies or swap out entirely). It might be better to provide an adapter interface for other HTTP client libraries.

**Comment by dentarg** [comment]
I guess this, if implemented, would expose the ability to change the User-Agent used by embiggen, right?

**Comment by mudge** [comment]
Yes: if we allowed users to access (and even provide) the HTTP client itself, then they could configure it however they wish (e.g. proxy settings).
